account/watcher/controller.go

```diff
@@ -161,34 +140,7 @@ func (c *controller) expiryHandler(blockChan chan int32, errChan chan error) {
 		// A new block notification has arrived, update our known
 		// height and notify any newly expired accounts.
 		case newBlock := <-blockChan:
-			bestHeight = uint32(newBlock)
-
-			for _, traderKey := range expirationsPerHeight[bestHeight] {
-				var accountKey [33]byte
-				copy(accountKey[:], traderKey.SerializeCompressed())
-
-				// If the account doesn't exist within the
-				// expiration set, then the request was
-				// canceled and there's nothing for us to do.
-				// Similarly, if the request was updated to
-				// track a new height, then we can skip it.
-				curExpiry, ok := expirations[accountKey]
-				if !ok || bestHeight != curExpiry {
-					continue
-				}
-
-				err := c.cfg.HandleAccountExpiry(
-					traderKey, bestHeight,
-				)
-				if err != nil {
-					log.Errorf("Unable to handle "+
-						"expiration of account %x: %v",
-						traderKey.SerializeCompressed(),
-						err)
-				}
-			}
-
-			delete(expirationsPerHeight, bestHeight)
+			c.watcher.NewBlock(uint32(newBlock))
```

Comment:
One thing to keep in mind is that if `HandleAccountExpiry` blocks for some time, then the `blockChan` might get clogged.